internal/clients/content_sources/client.go

```diff
@@ -47,7 +48,10 @@ func (csc *ContentSourcesClient) request(method, url string, headers map[string]
 		req.Header.Add(k, v)
 	}
 
-	return csc.client.Do(req)
+	// TODO context must be passed to the request method
+	req = req.WithContext(context.TODO())
+	doer := strc.NewTracingDoer(csc.client)
+	return doer.Do(req)
```

Comment:
Shall we split the tracing out to a separate PR here? Makes it easier to review/merging this one.